### ISSUE TITLE
Trezor connection status icon not working properly

### ISSUE THREAD

**Issue by lunarovic** [spesmilo/electrum#6569]
I opened Electrum with Trezor. There is this icon in the bottom-right corner: ![image](https://user-images.githubusercontent.com/2381553/92745564-c84cdd00-f382-11ea-9ebb-e9378652b0ce.png) Then I disconnected Trezor from PC. The icon remained the same, with the green dot. I clicked the icon to re-check Trezor connection, which I thought will maybe trigger the icon change. It displayed an error "device not found", but the green dot remained. I expected a red dot.

**Comment by SomberNight** [comment]
what version of Electrum?
https://github.com/spesmilo/electrum/commit/53a5a21ee89b791c351ad39a47e69837ed347af8 would fix the specific reproduction example you gave

However, even still, in general the status icon is not very responsive.
It's more complex than it seems at first though, one has to think about what the icon is supposed to signal.
Currently it signals not whether the device is connected; but whether the device is connected AND could be paired to the respective keystore in the wallet.

**Comment by lunarovic** [comment]
> what version of Electrum?

Both 3.3.8 and 4.0.2 behave the same for the reproduction steps above.

> 53a5a21 would fix the specific reproduction example you gave

Wow, 3 days ago, it didn't occur to me to check recent commits and find it's already fixed for the next version :)

> Currently it signals not whether the device is connected; but whether the device is connected AND could be paired to the respective keystore in the wallet.

I see. So if I got this correctly, in the next version, for my scenario the dot will turn red, because the device is not connected and therefore will not fulfill one of the two conditions? But it will also be red if it's connected but could not be paired to the respective keystore? Most importantly, the icon will become more responsive :)

**Comment by SomberNight** [comment]
Yes, the icon will become more responsive for the green->red direction. It will turn red at the next scan, regardless of reason for the scan (so in your example it would turn red when you click the button).

No changes for the red->green direction though. It will only turn green when an action requires pairing the keystore with the device (e.g. tx signing, or showing an address).
It's somewhat problematic to improve this; consider the edge case where the user has multiple wallets open that involve the device; if we pair it automatically we would have to choose to which one. So I think it's good enough as it is, and hence closing.
